File: leetcode/299.cc
#include <vector>
#include <string>


using namespace std;


class Solution {
  public:
    string getHint(string secret, string guess) {
        int n = secret.size();
        int bulls = 0;
        vector<int> cnt(10);
        for (int i = 0; i < n; ++i) {
            if (secret[i] == guess[i]) {
                ++bulls;
                secret[i] = ' ';
                guess[i] = ' ';
            } else {
                ++cnt[secret[i]-'0'];
            }
        }

        int cows = 0;
        for (int i = 0; i < n; ++i) {
            if (guess[i] == ' ')
                continue;
            if (cnt[guess[i]-'0'] > 0) {
                ++cows;
                --cnt[guess[i]-'0'];
            }
        }

        return to_string(bulls) + "A" + to_string(cows) + "B";
    }
};

File: leetcode/299.py
import string
from collections import Counter


class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        bulls = 0
        s_cnt = Counter()
        g_cnt = Counter()
        for s, g in zip(secret, guess):
            if s == g:
                bulls += 1
                continue
            s_cnt[s] += 1
            g_cnt[g] += 1
        cows = 0
        for d in string.digits:
            cows += min(s_cnt[d], g_cnt[d])
        return f'{bulls}A{cows}B'
